modules/library/main/src/main/java/org/geotools/filter/function/ParseTimeFunction.java

```
@@ -0,0 +1,71 @@
+ *
+ *    (C) 2025, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+import static org.geotools.util.DateTimeParser.FLAG_GET_TIME_ON_PRESENT;
+import static org.geotools.util.DateTimeParser.FLAG_IS_LENIENT;
+import static org.geotools.util.DateTimeParser.FLAG_SINGLE_DATE_AS_DATERANGE;
+
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.Date;
+import org.geotools.api.filter.capability.FunctionName;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.LiteralExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.geotools.util.DateRange;
+import org.geotools.util.DateTimeParser;
+
+/** Function parses GeoTools time notation into date. */
+public class ParseTimeFunction extends FunctionExpressionImpl {
+
+    public static FunctionName NAME = new FunctionNameImpl("parseTime", parameter("string", String.class));
+
+    DateTimeParser PARSER;
+
+    public ParseTimeFunction() {
+        super(NAME);
+        PARSER = new DateTimeParser(1, FLAG_GET_TIME_ON_PRESENT | FLAG_IS_LENIENT | FLAG_SINGLE_DATE_AS_DATERANGE);
+    }
+
+    @Override
+    public Object evaluate(Object object) {
+        Date dateTime = null;
+        try {
+            String value = (String) ((LiteralExpressionImpl) params.get(0)).getValue();
+            if (value.startsWith("P")) {
```

I see the convenience, but what if the user already provides "P1M/PRESENT"?
Maybe "if starts with P, and does not contain /" instead?
Something like "P1M" means "one month", it's a legal duration but not attached to a specific point in time. "P1M/2008-05-11T15:30:00Z" is the month before the final date, and so on. 
"P1M/PRESENT" is parsable in GT, but "PRESENT" has been added as a convenience; it's not part of the ISO spec.

Long story short, this convenience mode should be at least documented, because "P1M" alone is just a duration without any time reference (so using it to intend a relative time in the past, is a deviation from the spec).

---

Updated doc to mention that only single token is being used, and full range should not be used.
Allowing user to provide full range (e.g. P1M/PRESENT) can give a false impression that the second token affects processing somehow.

---

Yes, but... you did not say anything about a case like "2008-05-11T15:30:00Z/2008-05-11T18:30:00Z", which would be parsed in a legit way and just the first item be used.  Also, DateTimeParser can parse a comma- separated list of values, in which case it's going to return a List. If the user tries that, the code as written now will silently return null (should throw an exception, if one is using a syntax that is not mean to be supported).

---

> Yes, but... you did not say anything about a case like "2008-05-11T15:30:00Z/2008-05-11T18:30:00Z", which would be parsed in a legit way and just the first item be used. 

Will update doc to mention that.

> Also, DateTimeParser can parse a comma- separated list of values, in which case it's going to return a List

Can you provide an example of such list? I have checked the doc on parse method that mentions:
The string may contains either a single date, or a start time, end time and a period.

Also checked the DateTimeParserTest, but did not notice comma separated lists. Only ones separated by slashes (2011-10-10T10:11:12.000Z/2011-10-10T14:11:12.000Z/PT1H). This I believe can be covered by doc (saying that only single tokens are allowed).